db/seeds.rb

```diff
@@ -91,97 +91,110 @@ def attach_specific_school_to_organisation_if_present(organisation:, urn:)
   Location.where(urn:).update_all(team_id: organisation.generic_team.id)
 end
 
-def get_location_for_session(organisation, programme)
+def get_location_for_session(organisation, programmes)
+  year_groups = programmes.flat_map(&:year_groups).uniq
   loop do
     location =
-      organisation.locations.for_year_groups(programme.year_groups).sample ||
-        FactoryBot.create(
-          :location,
-          :school,
-          organisation:,
-          year_groups: programme.year_groups
-        )
+      organisation.locations.for_year_groups(year_groups).sample ||
+        FactoryBot.create(:location, :school, organisation:, year_groups:)
 
     return location unless organisation.sessions.exists?(location:)
   end
 end
 
-def create_session(user, organisation, completed:)
-  programme = Programme.find_by(type: "hpv")
-
-  FactoryBot.create_list(
-    :batch,
-    4,
-    organisation:,
-    vaccine: programme.vaccines.active.first
-  )
+def create_session(
+  user,
+  organisation,
+  programmes:,
+  completed: false,
+  year_groups: nil
+)
+  year_groups ||= programmes.first.year_groups
```

Comment:
I think we can get the year groups from all the programmes like above:

```suggestion
  year_groups ||= programmes.flat_map(&:year_groups).uniq
```